core/io/json.cpp

```diff
@@ -71,14 +71,16 @@ String JSON::_stringify(const Variant &p_var, const String &p_indent, int p_cur_
 			return itos(p_var);
 		case Variant::FLOAT: {
 			double num = p_var;
-			if (p_full_precision) {
-				// Store unreliable digits (17) instead of just reliable
-				// digits (14) so that the value can be decoded exactly.
-				return String::num(num, 17 - (int)floor(log10(num)));
-			} else {
-				// Store only reliable digits (14) by default.
-				return String::num(num, 14 - (int)floor(log10(num)));
+
+			if (num == double(0)) {
```

Comment:
Yeah, in this case I wanted it to be very clear that this was just and only dealing with the "Exactly 0" case, because otherwise the new could would return it as "0.00000000000000", but for "approximately 0" we don't want that, because if someone wants to serialize an "approximately 0" value, who are we to truncate it?